compiler/src/dotty/tools/dotc/typer/Checking.scala

```diff
@@ -206,7 +206,9 @@ object Checking {
           def isInteresting(prefix: Type): Boolean = prefix.stripTypeVar match {
             case NoPrefix => true
             case prefix: ThisType => sym.owner.isClass && prefix.cls.isContainedIn(sym.owner)
-            case prefix: NamedType => !prefix.symbol.isStaticOwner && isInteresting(prefix.prefix)
+            case prefix: NamedType =>
+              !prefix.symbol.isStaticOwner &&
+              (pre.derivesFrom(sym.owner) || isInteresting(prefix.prefix))
```

Comment:
In this case we are asking if the prefix `Foo.A` is interesting. It is as `Foo` extends `Base` (where `A` is defined).